Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -127,10 +127,49 @@ namespace PlayerCommands
 		pub::Player::PopUpDialog(client, caption, message, POPUPDIALOG_BUTTONS_CENTER_OK);
 	}
 
+	bool RateLimitLogins(uint client, PlayerBase *base, wstring charname)
+	{
+		uint curr_time = (uint)time(0);
+		uint big_penalty_time = 300;
+		uint amount_of_attempts_to_reach_penalty = 15;
+
+		//initiate
+		if (base->unsuccesful_logs_in_a_row.find(charname) == base->unsuccesful_logs_in_a_row.end())
+			base->unsuccesful_logs_in_a_row[charname] = 0;
+
+		if (base->last_time_of_log.find(charname) == base->last_time_of_log.end())
+			base->last_time_of_log[charname] = 0;
+
+		//nulify counter if more than N seconds passed.
+		if (curr_time - base->last_time_of_log[charname] > big_penalty_time)

```

Comment:
let's put brackets in here to make explicit what we mean please

Reply:
ok